src/drag_input.js

```diff
@@ -24,9 +24,95 @@ class gltfDragInput
     {
         event.preventDefault();
 
-        let additionalFiles = [];
+        let files = [];
+        let folders = [];
+        let droppedFiles = event.dataTransfer.files;
+        let droppedItems = event.dataTransfer.items;
+
+        for (let i = 0; i < droppedItems.length; i++)
+        {
+            let entry;
+            if (droppedItems[i].getAsEntry)
+            {
+                entry = droppedItems[i].getAsEntry();
+            }
+            else if (droppedItems[i].webkitGetAsEntry)
+            {
+                entry = droppedItems[i].webkitGetAsEntry();
+            }
+            if (!entry)
+            {
+                files.push(droppedFiles[i]);
+            }
+            else
+            {
+                if (entry.isDirectory)
+                {
+                    folders.push(entry)
+                }
+                else
+                {
+                    files.push(droppedFiles[i]);
+                }
+            }
+        }
+
+        if (folders.length === 0)
+        {
+            this._processFiles(files)
+        }
+        else
+        {
+            let remaining = folders.length;
+            for (let i = 0; i < folders.length; i++)
+            {
+                this._traverseFolder(folders[i], files, remaining, function(object)
+                {
+                    object._processFiles(files)
+                })
+            }
+        }
+    }
+
+    _traverseFolder(folder, files, remaining, callback)
```

Comment:
This underscore for fake privateness is not a convention we use, but I don't see any harm in it.